### DISCUSSION TITLE
MacOS group_commit_mgr.cpp编译错误

### DISCUSSION THREAD

**Discussion by peterylh** [apache/doris#36462] [A - General / Q&A]
最新的 Master 分支
编译器为ccache /usr/local/opt/llvm@16/bin/clang++
group_commit_mgr.cpp文件在Mac （macOS sonoma 14.5 版本）下编译出现问题 。
group_commit_mgr.cpp的152 行， 将常量10000L修改为10000LL，则可以编译成功。
大家有遇到同样问题吗， 是环境问题，还是代码有问题？

` 
 _get_cond.wait_for(l, std::chrono::milliseconds(std::min(left_milliseconds, 10000L)));
改为
  _get_cond.wait_for(l, std::chrono::milliseconds(std::min(left_milliseconds, 10000LL)));

`

出错下信息如下：
/xxxxx/doris/be/src/runtime/group_commit_mgr.cpp:152:57: error: no matching function for call to 'min'
        _get_cond.wait_for(l, std::chrono::milliseconds(std::min(left_milliseconds, 10000L)));
                                                        ^~~~~~~~
/usr/local/opt/llvm@16/bin/../include/c++/v1/__algorithm/min.h:40:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('int64_t' (aka 'long long') vs. 'long')
min(const _Tp& __a, const _Tp& __b)
^
/usr/local/opt/llvm@16/bin/../include/c++/v1/__algorithm/min.h:51:1: note: candidate template ignored: could not match 'initializer_list<_Tp>' against 'int64_t' (aka 'long long')
min(initializer_list<_Tp> __t, _Compare __comp)
^
/usr/local/opt/llvm@16/bin/../include/c++/v1/__algorithm/min.h:60:1: note: candidate function template not viable: requires single argument '__t', but 2 arguments were provided
min(initializer_list<_Tp> __t)
^
/usr/local/opt/llvm@16/bin/../include/c++/v1/__algorithm/min.h:31:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
min(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
1 error generated.

**Comment by ixzc** [comment]
中文问题，建议去ask.selectdb.com去问问题
